core/lib/zksync_core/src/lib.rs

```diff
@@ -496,11 +496,16 @@ pub async fn initialize_components(
             .build()
             .await
             .context("failed to build eth_watch_pool")?;
+        let governance = match (governance_contract(), contracts_config.governance_addr) {
```

Comment:
I'm worried about misconfiguration.

If contracts_config.governance_addr is set, we MUST have governance_contract() code available, right?

so maybe we should do:
if contracts_confg.governance_addr {
    governance = governance_contract.expect("governance contract must be present if governance_addr is set in config")
}

Reply:
I was thinking about such case 
- we deploy governor (e.g. for stage) before upgrade
- put its address in config
- don't update the server yet, so `governance_contract()` still returns `None`.

Then your approach would panic. But most probably we won't add governor_addr to config before updating server to version with post-boojum contracts. So, I don't have a strong opinion here